src/v/resource_mgmt/storage.cc

```diff
@@ -132,4 +132,8 @@ ss::future<> disk_space_manager::run_loop() {
     }
 }
 
+bool disk_space_manager::should_block_writes() const {
```

Comment:
I'm not sure it makes sense to reject produce requests in this scenario: we already reject produce requests when disk is critically low.  So this path would be saying "even though disk is not critically low, start blocking produce requests to keep the raft storage below its target".

Assuming that the target is achievable via truncation of local logs, it shouldn't be necessary to block producers.  And if the target _isn't_ achievable via truncation of local logs, _but_ there is enough disk space to safely proceed, I think the best behavior is to accept the produce requests.

Reply:
So I guess I'm proposing that we use some contextual knowledge: we know that the reason for the target size existing in the first place is disk space management, and that it exists to protect free space on a shared drive from being eaten up by very long local raft retention.

Reply:
Blocking produce was making me uneasy to begin with, and the argument here for favoring a, "sloppy" isn't the right word, but delayed accounting to avoid the nuclear option of blocking makes sense. Things will be more robust.